faithcomesbyhearing/sofria-cli
# FCBHDBP-515 enable non-integer verse identifier - sofria

## Description
The handler that generates verses and stores them in the SQLite database has been modified to support the scenario where a chapter contains multiple verses with the same number but different categories (e.g. 6a and 6b). Previously, only the first verse was stored.

## Issue Link
https://fullstacklabs.atlassian.net/browse/FCBHDBP-515

## How Do I QA This
- You should run the following tests and they pass:
- You should run the following tests and they pass:
```
node tests/gqlquery_verses_by_usx.test.js
```
- outcome
```
TAP version 13
# Verse Rows From Chapters test
ok 1 should be strictly equal
ok 2 should be strictly equal
ok 3 should be strictly equal
# Verse Rows To Insert
ok 4 should be strictly equal
ok 5 should be strictly equal
ok 6 should be strictly equal
# Missed Verse test
ok 7 should be strictly equal
# Missed Range Verse diff = 1 test
ok 8 should be strictly equal
ok 9 should be strictly equal
ok 10 should be strictly equal
ok 11 should be strictly equal
ok 12 should be strictly equal
# Missed Verse Range diff > 1 test
ok 13 should be strictly equal
ok 14 should be strictly equal
ok 15 should be strictly equal
ok 16 should be strictly equal
ok 17 should be strictly equal
# Same verse number but with different sub-category a and b
ok 18 should be strictly equal
ok 19 should be strictly equal
ok 20 should be strictly equal
ok 21 should be strictly equal
ok 22 should be strictly equal

1..22
# tests 22
# pass  22

# ok
```
- Also, you can run the following command and they should return success:
```
node biblebrain_uploader.js run test/input/TXQWYIN1ET/ --populate-db=./output/TXQWYIN1ET-db/TXQWYIN_ET.db --generate-json=./output/TXQWYIN_ET-json
```

- Outcome
```
Generate JSON => Generate JSON filese - Start process..
Populate DB - Start process..
Populate DB => Complete:  Markus
Populate DB => success
Generate JSON => File list (1) processing completed
Generate JSON => Complete USX File: test/input/TXQWYIN1ET/041MRK.usx
load verses success, rowcount 634
load tableContents success, rowcount 1

```
